projects/table-builder/src/lib/pipes/phone.pipe.ts

```diff
@@ -0,0 +1,14 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({name: 'phone'})
+export class PhoneNumberPipe implements PipeTransform {
+
+  transform(phoneNum: string): any {
+    if (phoneNum) {
+      if(phoneNum[0] === '1') {
+        phoneNum = phoneNum.substring(1);
+      }
+      return '(' +  phoneNum.slice(0, 3) + ') ' + phoneNum.slice(3, 6) + '-' + phoneNum.slice(6, 10) + ' ' + phoneNum.slice(10);
```

Comment:
what if the characters are already there?

Reply:
i thought about that. not sure how you want to deal with it, i thought maybe if you have such data then just use field type string like the old way. I could always clean the string first, and then add the stuff?